test/base/test_objective.py

```diff
@@ -251,10 +269,14 @@ def test_fds(fd_method):
         val_fd_grad = getattr(obj_fd_grad, f"get_{attr}")(p)
         val_fd_fake = getattr(obj_fd_fake, f"get_{attr}")(p)
 
-        for v in [val_fd, val_fd_grad]:
-            assert np.allclose(val, v, atol=atol, rtol=rtol), attr
-            # cannot completely coincide
-            assert (val != v).any(), attr
+        assert np.allclose(val, val_fd, atol=atol, rtol=rtol), attr
```

Comment:
Can we maybe double check, whether we achieve better accuracies when using adaptive FDs than when using static FDs? Maybe even down to 1e-5?

Reply:
The default is 1e-6, which already gives quite a good approximation (automatically chosen are 1e-7 to 1e-8), so on this problem not much gain, rather a "automatic detection does not screw things up" check.